test_runner/regress/test_s3_restore.py

```diff
@@ -0,0 +1,111 @@
+import time
+from datetime import datetime, timezone
+
+from fixtures.neon_fixtures import (
+    NeonEnvBuilder,
+    PgBin,
+    wait_for_upload,
+)
+from fixtures.pageserver.utils import (
+    MANY_SMALL_LAYERS_TENANT_CONFIG,
+    assert_prefix_empty,
+    poll_for_remote_storage_iterations,
+    tenant_delete_wait_completed,
+)
+from fixtures.remote_storage import s3_storage
+from fixtures.utils import run_pg_bench_small
+from fixtures.types import Lsn
+
+
+def test_tenant_s3_restore(
```

Comment:
Maybe another follow-up: doing the restore for sharded tenant. Otherwise there might be some last-minute changes and they'd break without us being aware of them.

Reply:
resolving as this is a planned followup (https://github.com/neondatabase/neon/pull/6533#issuecomment-1919587149)